practice/vue-app/src/module2/rest-countries/services/AxiosService.ts

```diff
@@ -0,0 +1,51 @@
+import axios, { isAxiosError } from 'axios';
+
+export class AxiosService implements HttpService {
+  private httpService;
+
+  constructor(baseURL: `https://${string}`) {
+    this.httpService = axios.create({ baseURL: baseURL });
+    this.interceptResponse();
+  }
+
+  interceptResponse = () => {
+    this.httpService.interceptors.response.use(
+      (response) => {
+        console.log('Incoming response', response);
+        return response;
+      },
+      (error) => {
+        if (isAxiosError(error)) {
+          console.error(`Axios error intercepted`, error);
+        } else {
+          console.error('General error intercepted:', error);
+        }
+        return Promise.reject(error);
+      }
+    );
+  };
+
+  getAll = async () => {
+    try {
+      const response = await this.httpService.get('all');
+      return response.data;
+    } catch (error) {
+      if (isAxiosError(error) && error.response) {
+        console.log(error.response?.data);
+      }
+      return [];
+    }
+  };
+
+  filter = async (path: string) => {
+    try {
+      const response = await this.httpService.get(path);
+      return response.data;
+    } catch (error) {
+      if (isAxiosError(error) && error.response) {
+        console.log(error.response?.data);
+      }
+      return [];
+    }
+  };
+}
```

Comment:
Mogłbyś sobie dorzucić jeszcze:

```
private handleError(error: unknown): void {
    if (isAxiosError(error) && error.response) {
      console.error('Response error data:', error.response.data);
    }
}
```

Uniknąłbyś tym samym powtórzeń w metodach `getAll` i `filter`.

Idąc dalej, mógłbyś utworzyć dodatkowo metodę:
```
private async handleRequest<T>(path: string): Promise<T[]> {
    try {
      const response = await this.httpService.get<T[]>(path);
      return response.data;
    } catch (error) {
      this.handleError(error);
      return [];
    }
}
```
  
  To z kolei pozwoliłoby uniknąć powtórzeń pozostałego kodu z metod `getAll` i `filter`